enhancements/network/user-space-pod-iface-and-api-library.md

```diff
@@ -0,0 +1,262 @@
+of APIs. The API is written in Golang and can be imported into other Golang
+programs. Since DPDK is the primary user space application being targeted, and
+DPDK is written in C, there is also a C API binding for the Golang library.
+
+The library is located in github at the following URL:
+- https://github.com/openshift/app-netutil/
+
+The library will not be released in OCP, but will be available for CNF
+developers to include in their projects. If the CNF developers choose not to
+include the library, the library will still serve as an example of where to pull
+the data from and how to consume the data.
+
+The following APIs will be exposed:
+- GetCPUInfo()
+  - Returns the set of CPUs assigned to the pod.
+- GetInterfaces()
+  - Returns the set of interfaces associated with the pod and a structure for
+  each interface containing the type of interface and the data specific to that
+  interface type.
+- GetHugepageInfo()
+  - Returns the amount of hugepage memory reserved for the pod and page size.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+### Graduation Criteria
+
+The '*appNetUtil*' library was introduced as Tech Preview in OCP 4.5. The plan
+is to make the library GA in OCP 4.7. To our knowledge, the '*appNetUtil*'
+library is not currently being used. With the additions described in this
+enhancement, the library should be more useful to CNF developers. Also by moving
+to GA, the library attract more users and provide feedback on ways to improve
+the usefulness  of the library.
+
+##### Dev Preview -> Tech Preview
+
+In Tech Preview, the following APIs are defined in
+"*github.com/openshift/app-netutil/lib/v1alpha*":
+- GetCPUInfo()
+  - Returns the set of CPUs assigned to the pod.
+- GetInterfaces()
+  - Returns the set of interfaces associated with the pod and a structure for
+  each interface containing the type of interface and the data specific to that
+  interface type.
+
+##### Tech Preview -> GA 
+
+In GA, the following APIs are defined in
+"*github.com/openshift/app-netutil/lib/v1*":
+
+- GetCPUInfo()
+  - Returns the set of CPUs assigned to the pod.
+  - Same as Tech Preview.
+- GetInterfaces()
+  - Returns the set of interfaces associated with the pod and a structure for
+  each interface containing the type of interface and the data specific to that
+  interface type.
+  - The returned structure will be expanded to include additional data.
+- GetHugepageInfo()
+  - Returns the amount of hugepage memory reserved for the pod and page size.
+  - New API.
+
+##### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+## Implementation History
+
+appNetUtil Library:
+- Tech Preview: OCP 4.5 - June, 2020
+- GA: OCP 4.7 - February, 2021
+
+## Drawbacks
+
+## Alternatives
+
+## Infrastructure Needed [optional]
```

Comment:
Integrate the app-netutil project with CI ?

Reply:
Are you asking about CI within the https://github.com/openshift/app-netutil/ (which I think we need to add) or integrated into the OCP CI?

Reply:
CI in app-netutil repo.

Reply:
CI in app-netutil repo.

Reply:
Added a CI note to the Test Plan section.